session/session_test.go

```diff
@@ -48,11 +52,60 @@ func TestReceiveEventWhenSubscribing(t *testing.T) {
 		if event != SessionDisconnected {
 			t.Error("Expected to receive disconnected: ", event)
 		}
+	case <-time.After(5 * time.Second):
+		t.Error("Failed to receive event")
+	}
 
-		if <-events != SessionReconnected {
+	select {
+	case event := <-events:
+		if event != SessionReconnected {
 			t.Error("Expected to receive reconnected: ", event)
 		}
-	case <-time.After(100 * time.Millisecond):
+	case <-time.After(5 * time.Second):
 		t.Error("Failed to receive event")
 	}
 }
+
+func TestResumeZKSessionWithValidSession(t *testing.T) {
+	proxy := CreateProxy(t)
+	defer proxy.Delete()
+
+	store, err := NewZKSession("localhost:27445", 200*time.Millisecond, nil)
+	if err != nil {
+		t.Error("Failed to connect to Zookeeper: ", err)
+	}
+
+	existingClientId, err := store.ClientId().Save()
+	if err != nil {
+		t.Error("Failed to save clientId: ", err)
+	}
+
+	store.Delete("localhost:27445", -1)
```

Comment:
It might be worth adding a 'quit' channel to the `manage()` function so that the goroutine won't reconnect in the background.  This would enable us to send a quit in this test, and hence properly test `ResumeZKSession` after removing the toxiproxy toxic.  Though in practice, this quit channel is unlikely to be used otherwise; what we're really trying to accomplish is simulating a process restart (and the goroutines quitting)